packages/google_maps_flutter/lib/src/google_map.dart

```diff
@@ -472,6 +480,10 @@ class _GoogleMapOptions {
     addIfNonNull('scrollGesturesEnabled', scrollGesturesEnabled);
     addIfNonNull('tiltGesturesEnabled', tiltGesturesEnabled);
     addIfNonNull('zoomGesturesEnabled', zoomGesturesEnabled);
+    addIfNonNull('liteModeEnabled', liteModeEnabled);
+    if (liteModeEnabled != null && liteModeEnabled && !Platform.isAndroid) {
+      print('Warning: liteModeEnabled is Android only');
```

Comment:
Instead of printing a warning message, what do you think about adding an assert(with warning message) here so developers can encounter a crash in debug mode. This way we are sure the developers know that this is android only.

Reply:
Thus if developers knows about this behaviour and consciously want to use maps on Android with enabled light mode and disabled on iOS they should write platform specific code in their app to avoid crashes in debug mode.

Something like this:

```dart
Map(
  liteModeEnabled: Platform.isAndroid,
);
```

Is this the right approach to working with plugins?

Reply:
+1 @sanekyy